### PR TITLE
Antu3199/todos

### PR DESCRIPTION
**Pull request by antu3199**
- Selection targeting outline effect
- Mooks joining mid-battle
- Hero can now use magic!

### CODE REVIEW

File: Assets/_Scripts/StageInfo.cs [hunk 1]
```diff
@@ -29,45 +24,78 @@ public class StageInfo : MonoBehaviour
     public Player GetHeroPlayer() {
         return _heroPlayer;
```
Comment (BenjixD) [inline Assets/_Scripts/StageInfo.cs:25]: Do we really need reference to hero player here? We already have reference in party.

File: Assets/_Scripts/TurnController.cs [hunk 2]
```diff
@@ -189,29 +205,82 @@ void Update()
     private void setHeroAction() {
         _heroActionIndex = this.commandSelector.GetChoice();
```
Comment (BenjixD) [inline Assets/_Scripts/TurnController.cs:206]: We should eventually pull this out from TurnController and create a separate HeroInput object.

File: Assets/GameManager.cs [hunk 3]
```diff
@@ -37,36 +37,36 @@ public class GameManager : Singleton<GameManager> {
         party.CreateHeroPlayer();
     }
 
-    public List<ActionBase> GetPlayerJobActionsList(Job job) {
+    public List<ActionBase> GetPlayerJobActions(Job job) {
         if(!playerJobActions.ContainsKey(job)) {
```
[the remark this answers is not in the document]
Comment (antu3199) [reply Assets/GameManager.cs:41]: The reason for this is that if I wanted to instantiate a random player, or a random enemy, then I need to know which job corresponds to which list. Alternatively you can just put a flag in the JobActionList.

File: Assets/_Scripts/Party.cs [hunk 4]
```diff
@@ -4,20 +4,21 @@
...
-    [SerializeField]
-    private PlayerCreationData[] playerPos = new PlayerCreationData[4];
+    private PlayerCreationData[] playerPos = new PlayerCreationData[numPlayers];
 
     public void CreatePlayer(PlayerCreationData data, int index) {
         playerPos[index] = data;
     }
 
     public void CreateHeroPlayer() {
         // Optional TODO: If streamer name is special, then give them a special hero
```
[the remark this answers is not in the document]
Comment (antu3199) [reply Assets/_Scripts/Party.cs:19]: I think it's fine either way.  Remember that GameManager is usually for things where we don't know where else to put it.

File: Assets/_Scripts/StageInfo.cs (hunk 1 above)
Comment (antu3199) [reply Assets/_Scripts/StageInfo.cs:25]: It's just for shorthand

File: Assets/_Scripts/TurnController.cs (hunk 2 above)
Comment (antu3199) [reply Assets/_Scripts/TurnController.cs:206]: I'm fine either way, but managing the input is the only unique thing that the TurnController does right now. The rest is done in Battle or BattleFight

**Pull request merged**